src/thread_pool.rs

```diff
@@ -903,7 +901,7 @@ pub struct WorkerThread {
 }
 
 thread_local! {
-    static WORKER_THREAD_STATE: Cell<*const WorkerThread> = const { Cell::new(ptr::null()) };
+    static WORKER_THREAD_STATE: CachePadded<OnceCell<WorkerThread>> = const { CachePadded::new(OnceCell::new()) };
```

Comment:
This is so that atomics in the static memory section don't interfere with reads? Clever, I missed this.

Reply:
I didn't actually test it, so I don't know whether it's necessary :).  The old code was definitely fine, because it stored these on the stack.  I read something recently about how the `thread_local` crate [can suffer from false sharing](https://morestina.net/blog/1976/a-close-encounter-with-false-sharing), but I have no idea whether ordinary `thread_local!`s would have a similar issue.

Reply:
I think this is probably needed here.